fs-mock.js

```diff
@@ -126,7 +126,9 @@ MockFs.prototype.open = function (path, flags, charset, options) {
                     charset
                 );
             } else {
-                return new BufferStream(fileNode._chunks, charset);
+                // Clone chunks to avoid side effect
+                var bufferChunks = fileNode._chunks.slice(0);
```

Comment:
Idiom is slice() without the implicit 0 arg.

Could also add a BufferStream.prototype.clone() to do less reaching through the encapsulation.